### PR TITLE
Add logs to EDL

### PR DESCRIPTION
**Pull request by ilaner**
This adds extensive logging to the configuration, and adds new request parameter for the indicators after the search and before the format.

related: https://jira-dc.paloaltonetworks.com/browse/XSUP-32470
related: https://jira-dc.paloaltonetworks.com/browse/XSUP-31635

### CODE REVIEW

File: Packs/EDL/Integrations/EDL/EDL.yml [hunk 1]
```diff
@@ -262,6 +262,14 @@ configuration:
   section: Collect
   advanced: true
   required: false
+- name: extensive_logging
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.yml:265]: looks like display field is missing.
Should be "Extensive Logging"

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 2]
```diff
@@ -381,8 +384,8 @@ def get_indicators_to_format(indicator_searcher: IndicatorsSearcher,
             # NG + XSIAM can recover from a shutdown
             if version.get('platform') == 'x2' or is_demisto_version_ge('8'):
                 raise SystemExit('Encountered issue in Elastic Search query. Restarting container and trying again.')
-
-    demisto.debug(f"Completed IOC search & format, found {ioc_counter} IOCs.")
+    if extensive_logging:
+        demisto.debug(f"edl: Completed IOC search & format, found {ioc_counter} IOCs. Their contents: {indicators}")
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.py:388]: in a separate log line, please print the length of indicators or ioc counter.
this should be printed regardless of extensive_logging

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 3]
```diff
@@ -760,13 +763,21 @@ def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO
             new_line = '\n'
     iocs.close()
     if len(ipv4_formatted_indicators) > 0:
+        if extensive_logging:
+            demisto.debug(f"edl: Original IPv4 indicators: {ipv4_formatted_indicators}")
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.py:767]: in a separate log line, please print the length of Original ipv4_formatted_indicators.
this should be printed regardless of extensive_logging

also, is it original if the var called formatted?

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 4]
```diff
@@ -760,13 +763,21 @@ def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO
...
         ipv4_formatted_indicators = ips_to_ranges(ipv4_formatted_indicators, request_args.collapse_ips)
+        if extensive_logging:
+            demisto.debug(f"edl: Formatted IPv4 indicators: {ipv4_formatted_indicators}")
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.py:770]: in a separate log line, please print the length of Formated ipv4_formatted_indicators.
this should be printed regardless of extensive_logging

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 5]
```diff
@@ -760,13 +763,21 @@ def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO
...
         for ip in ipv4_formatted_indicators:
             formatted_indicators.write(new_line + str(ip))
             new_line = '\n'
 
     if len(ipv6_formatted_indicators) > 0:
+        if extensive_logging:
+            demisto.debug(f"edl: Original IPv6 indicators: {ipv6_formatted_indicators}")
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.py:777]: the same addition of longline of length like the comments above

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 6]
```diff
@@ -760,13 +763,21 @@ def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO
...
         ipv6_formatted_indicators = ips_to_ranges(ipv6_formatted_indicators, request_args.collapse_ips)
+        if extensive_logging:
```
Comment (yuvalbenshalom) [inline Packs/EDL/Integrations/EDL/EDL.py:779]: the same addition of longline of length like the comments above

**Review by yuvalbenshalom — COMMENTED** [review]
see my comments

File: Packs/EDL/Integrations/EDL/EDL.yml (hunk 1 above)
Comment (ilaner) [reply Packs/EDL/Integrations/EDL/EDL.yml:265]: added

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 2 above)
Comment (ilaner) [reply Packs/EDL/Integrations/EDL/EDL.py]: added

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 3 above)
Comment (ilaner) [reply Packs/EDL/Integrations/EDL/EDL.py:774]: yes, it is before the format

File: Packs/EDL/Integrations/EDL/EDL.py (hunk 5 above)
Comment (ilaner) [reply Packs/EDL/Integrations/EDL/EDL.py:786]: done

File: Packs/EDL/Integrations/EDL/EDL.py [hunk 7]
```diff
@@ -760,13 +770,25 @@ def create_text_out_format(iocs: IO, request_args: RequestArguments) -> Union[IO
             new_line = '\n'
     iocs.close()
     if len(ipv4_formatted_indicators) > 0:
+        demisto.debug(f"edl: length of ipv4_formatted_indicators: {len(ipv4_formatted_indicators)}")
+        if extensive_logging:
+            demisto.debug(f"edl: Original IPv4 indicators: {ipv4_formatted_indicators}")
```
Comment (MLainer1) [inline Packs/EDL/Integrations/EDL/EDL.py:775]: What is the difference between both of those debug logs?

**Review by MLainer1 — COMMENTED** [review]
Nice!
Make sure to add the new request parameter to the integration description.

**Pull request closed without merging**